### PR TITLE
Upgrade to API v2

### PR DESCRIPTION
**Pull request by dfalmeida**
## Description
- Updated API to v2
- Updated packages dependencies to latest versions
- Updated README
- Do not run snapshot testing on CI to avoid comparison errors
- Changed the iOS test device from `iPhone 12 Pro Max` to `iPhone 13`

### CODE REVIEW

File: PokemonCards/Application/Cards/CardsCore.swift [hunk 1]
```diff
@@ -115,21 +115,20 @@ let cardsReducer =
...
+          ))
+        }
+
         return .concatenate(
           .init(value: .loadingActive(false)),
           .init(value: .loadingPageActive(false))
         )
 
       case .cardsResponse(.failure(let error)):
+        print(error)
```
Comment (andreragsilva) [inline PokemonCards/Application/Cards/CardsCore.swift]: Should we keep debug code? 🤔

**Review by andreragsilva — CHANGES_REQUESTED** [review]
@dfalmeida it also appears you changed the snapshot test screenshots by lowering the resolution. Was the high resolution affecting testing? This may affect the level of interface testing we have in terms of pixel perfect hence the curiosity 🤔

**Comment by andreragsilva** [comment]
> @dfalmeida it also appears you changed the snapshot test screenshots by lowering the resolution. Was the high resolution affecting testing? This may affect the level of interface testing we have in terms of pixel perfect hence the curiosity 🤔

Also, tests are failing as you mentioned before, just leaving a heads up 👍

**Comment by andreragsilva** [comment]
@dfalmeida CI is still failing 🤔  So I will wait till there is time to fix it so I can review once again 👍

**Comment by andreragsilva** [comment]
@dfalmeida CI still failing so I will postpone my review once again, hope you are able to close this 💪 Let me know if you need help

File: PokemonCards/Application/Cards/CardsCore.swift (hunk 1 above)
Comment (dfalmeida) [reply PokemonCards/Application/Cards/CardsCore.swift]: Sure not! 😓

**Comment by dfalmeida** [comment]
> @dfalmeida it also appears you changed the snapshot test screenshots by lowering the resolution. Was the high resolution affecting testing? This may affect the level of interface testing we have in terms of pixel perfect hence the curiosity 🤔

I changed the test device to a smaller one, maybe that was the cause 🤔

**Comment by dfalmeida** [comment]
@andreragsilva just disabled the snapshot testing on CI. It is something that needs a deeper investigation. 
As far as I understood, the CI machines generates a pixelated snapshot that doesn't match the original one, I tried everything I could: 
* Changed to a different macOS version
* Changed the snapshot strategy
* Find a way to change the graphic requirements from the machine. Cannot find those settings ofc :/ 

In the end CI is green 🍏😑

**Review by andreragsilva — APPROVED** [review]
@dfalmeida lets close this, I believe when and if we pick this up again, a lot will have change so we will be able to find an easier path (hopefully) 👍

Great work anyway, hope you enjoyed it 💪

**Pull request merged**